app/src/main/java/org/fossasia/pslab/communication/sensors/AD7718.java

```diff
@@ -0,0 +1,326 @@
+        return spi.send16(val);
+    }
+
+    public int readRegister(int reg) throws IOException {
+        start();
+        int val = send16(0x4000 | (reg << 8));
+        stop();
+        val &= 0x00FF;
+        return val;
+    }
+
+    public int readData() throws IOException {
+        start();
+        int val = send16(0x4000 | (ADCDATA << 8));
+        val &= 0xFF;
+        val <<= 16;
+        val |= send16(0x0000);
+        stop();
+        return val;
+    }
+
+    public int writeRegister(int reg, int value) throws IOException {
+        start();
+        int val = send16((reg << 8) | value);
+        stop();
+        return val;
+    }
+
+    public void internalCalibration(int chan) throws IOException, InterruptedException {
+        start();
+        int val = send16((ADCCON << 8) | (chan << 4) | 7);  // range=7
+        long start_time = System.currentTimeMillis();
+        caldone = false;
+        val = send16((MODE << 8) | 4);
+        while (!caldone) {
+            Thread.sleep(500);
+            caldone = (send16(0x4000 | (MODE << 8)) & 7) == 1;
+            Log.d(TAG, String.format("Waiting for zero scale calibration... %.2f S, %s",
+                    (float) (System.currentTimeMillis() - start_time),
+                    caldone)
+            );
+        }
+        caldone = false;
+        val = send16((MODE << 8) | 5);
+        while (!caldone) {
+            Thread.sleep(500);
+            caldone = (send16(0x4000 | (MODE << 8)) & 7) == 1;
+            Log.d(TAG, String.format("Waiting for full scale calibration... %.2f S, %s",
+                    (float) (System.currentTimeMillis() - start_time),
+                    caldone)
+            );
+        }
+        stop();
+    }
+
+    public List readCalibration() throws IOException {
+        start();
+        int off = send16(0x4000 | (ADCOFFSET << 8));
+        off &= 0xFF;
+        off <<= 16;
+        off |= send16(0x0000);
+
+        int gn = send16(0x4000 | (ADCGAIN << 8));
+        gn &= 0xFF;
+        gn <<= 16;
+        gn |= send16(0x0000);
+        stop();
+        return Arrays.asList(new int[] {off, gn});
+    }
+
+
+    public void configADC(int adccon) throws IOException {
+        writeRegister(ADCCON, adccon); // unipolar channels, range
+        gain = 2 ^ (7 - adccon & 3);
+    }
+
+    public void printstat() throws IOException {
+        int stat = readRegister(STATUS);
+        String[] P = {"PLL LOCKED", "RES", "RES", "ADC ERROR", "RES", "CAL DONE", "RES", "READY"};
+        String[] N = {"PLL ERROR", "RES", "RES", "ADC OKAY", "RES", "CAL LOW", "RES", "NOT READY"};
+        StringBuilder sb = new StringBuilder();
+        for (int a = 0; a < 8; a++) {
+            if ((stat & (1 << a)) == 1) {
+                sb.append(P[a]);
+            } else {
+                sb.append(N[a]);
+            }
+        }
+        Log.d(TAG, stat + ", " + sb.toString());
+    }
+
+    public float convert_unipolar(float x) {
```

Comment:
Avoid using '_' in method or Variables name in java. should be "convertUnipolar" instead.